CryptoTechReminderSystem.Main/Program.cs

```diff
@@ -1,43 +1,97 @@
-using System.Collections.Generic;
+using System.Threading;
 using CryptoTechReminderSystem.Boundary;
 using CryptoTechReminderSystem.Gateway;
 using CryptoTechReminderSystem.UseCase;
 using dotenv.net;
+using FluentScheduler;
 
 namespace CryptoTechReminderSystem.Main
 {
-    class Program
+    class Clock : IClock
     {
-        static void Main(string[] args)
+        public DateTimeOffset Now()
+        {
+            return DateTimeOffset.Now;
+        }
+    }
+
+    public class Program : Registry
+    {
+        private static void Main(string[] args)
         {    
             DotEnv.Config(false);
-            
-            var remindDeveloper = new SendReminder(
-                new SlackGateway(
-                    "https://slack.com/",
-                    Environment.GetEnvironmentVariable("SLACK_TOKEN")
-                )
+            JobManager.Initialize(new ReminderRegistry());
+            SpinWait.SpinUntil(() => false);
+        }
+    }
+
+    public class ReminderRegistry : Registry
+    {
+        private readonly IClock _clock;
+        private readonly GetLateDevelopers _getLateDevelopers;
+        private readonly SendReminder _sendReminder;
+
+        public ReminderRegistry()
+        {
+            var slackGateway = new SlackGateway(
+                "https://slack.com/",
+                Environment.GetEnvironmentVariable("SLACK_TOKEN")
+            );
+            var harvestGateway = new HarvestGateway(
+                "https://api.harvestapp.com/",
+                Environment.GetEnvironmentVariable("HARVEST_TOKEN"),
+                Environment.GetEnvironmentVariable("HARVEST_ACCOUNT_ID")
             );
-            
-            IList<string> idsList = new List<string>()
-            {
-                "UH3DVNTH7", 
-                "UH9NYEWSC", 
-                "UHC4PEXM2", 
-                "UHCABC886", 
-                "UHCFC7Z61", 
-                "UHDJJA52S"
-            };
+            _clock = new Clock();
+            _getLateDevelopers = new GetLateDevelopers(slackGateway, harvestGateway, harvestGateway, _clock);
+            _sendReminder = new SendReminder(slackGateway);
 
-            foreach (var id in idsList)
-            {
-                remindDeveloper.Execute(new SendReminderRequest
+            CreateSchedule();
+        }
+
+        private void ResetSchedule()
+        {
+            // Stops scheduled jobs then sets them up again for next week
+            JobManager.RemoveAllJobs();
+            CreateSchedule();
+        }
+        
+        private void CreateSchedule()
+        {
+            // Creates schedule that starts and stops schedule jobs
+            JobManager.AddJob(ScheduleJobs, s => s.ToRunEvery(1).Weeks().On(DayOfWeek.Friday).At(10,30));
+            JobManager.AddJob(ResetSchedule, s => s.ToRunEvery(1).Weeks().On(DayOfWeek.Friday).At(13,45));
+        }
+
+        private void ScheduleJobs()
+        {
+            // Do all of your actual job scheduling here
```

Comment:
Unnecessary commenting

Reply:
Resolved in https://github.com/chubberlisk/crypto_tech/pull/25/commits/8432cfb2334ce8c2878f5084f4db4d2526bcc8c7.